src/sbt-test/sbt-js-engine/jstask/project/SbtHelloJsTask.scala

```diff
@@ -43,18 +43,22 @@ object SbtHelloJsTask extends AutoPlugin {
     ).toString()
   )
 
+  override def buildSettings = inTask(hello)(
+    SbtJsTask.jsTaskSpecificUnscopedBuildSettings ++ Seq(
+      moduleName := "hello",
+      shellFile := SbtHelloJsTask.getClass.getClassLoader.getResource("hello-shell.js")
+    )
+  )
+
   override def projectSettings = Seq(
     compress := false,
     fail := false
   ) ++
     inTask(hello)(
-      SbtJsTask.jsTaskSpecificUnscopedSettings ++
+      SbtJsTask.jsTaskSpecificUnscopedProjectSettings ++
         inConfig(Assets)(helloJsTaskUnscopedSettings) ++
         inConfig(TestAssets)(helloJsTaskUnscopedSettings) ++
         Seq(
-          moduleName := "hello",
-          shellFile := SbtHelloJsTask.getClass.getClassLoader.getResource("hello-shell.js"),
-
```

Comment:
This shows what an sbt-web plugin will need to do to get the fix, essentially, `moduleName`, `shellFile` and the new `jsTaskSpecificUnscopedBuildSettings` need to be set by `AutoPlugin.buildSettings`, while `AutoPlugin.projectSettings` needs to set `jsTaskSpecificUnscopedProjectSettings` and not `moduleName` or `shellFile`.